plugins/formal-verification/src/org/jetbrains/kotlin/formver/plugin/ViperPoweredDeclarationChecker.kt

```diff
@@ -74,4 +78,9 @@ class ViperPoweredDeclarationChecker(val session: FirSession) : FirSimpleFunctio
         verifier.start()
         return verifier
     }
+
+    private fun getProgramForLogging(program: Program): Program? = when (logLevel) {
```

Comment:
Depending on what we want to log or not it might make sense to make this method return a String instead of a Program. It's not more complicated but give us more flexibility for the future.

Reply:
I would prefer to defer that to the last moment; it will make debugging this function harder if we ever need to.  It's not too much trouble to change later if it turns out to be necessary.

Reply:
Alright, then we can leave it like this as long as we don't need the generality.